tests/cli/commands/test_pool.py

```diff
@@ -0,0 +1,119 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import json
+import os
+import unittest
+
+import pytest
+from click.testing import CliRunner
+
+from airflow import models, settings
+from airflow.cli.commands import pool
+from airflow.models import Pool
+from airflow.settings import Session
+from airflow.utils.db import add_default_pool_if_not_exists
+
+
+class TestCliPools(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.dagbag = models.DagBag(include_examples=True)
+        cls.runner = CliRunner()
+
+    def setUp(self):
+        super().setUp()
+        settings.configure_orm()
+        self.session = Session
+        self._cleanup()
+
+    def tearDown(self):
+        self._cleanup()
+
+    @staticmethod
+    def _cleanup(session=None):
+        if session is None:
+            session = Session()
+        session.query(Pool).filter(Pool.pool != Pool.DEFAULT_POOL_NAME).delete()
+        session.commit()
+        add_default_pool_if_not_exists()
+        session.close()
+
+    def test_pool_list(self):
+        self.runner.invoke(pool.pools, ["set", "foo", "1", "test"])
+        result = self.runner.invoke(pool.pools, ["list"])
+        assert "foo" in result.output
+
+    def test_pool_list_with_args(self):
+        result = self.runner.invoke(pool.pools, ["list", "--output", "json"])
```

Comment:
Some of these tests are interdependent/overlapping. The scope of this PR is "converting to click cli" so I will leave refactoring for another time.